src/codeu/chat/client/core/UserContext.java

```diff
@@ -53,4 +53,33 @@ public ConversationContext start(String name) {
 
     return all;
   }
+
+  public int addUserInterest(String name) {
+    return controller.addUserInterest(name, user.id);
+  }
+
+  public int removeUserInterest(String name) {
+    return controller.removeUserInterest(name, user.id);
+  }
+
+  public int addConversationInterest(String title) {
+    return controller.addConversationInterest(title, user.id);
+  }
+
+  public int removeConversationInterest(String title) {
+    return controller.removeConversationInterest(title, user.id);
+  }
+
+  public ArrayList<String> userStatusUpdate(String name) {
+    final ArrayList<String> contributions = new ArrayList<>();
+    for(final String contribution : view.userStatusUpdate(name, user.id)) {
```

Comment:
Just fixed this. I forgot that I meant to import Collection here and just return a Collection rather than moving everything into an ArrayList and returning that (because ArrayList was already imported).